### PR TITLE
[#24] [UI] As a user, I can see the NPS component in the Question page

### LINKED ISSUE #24: [UI] As a user, I can see the NPS component in the Question page
## Why
The user can see the questions and the corresponding answers in the Question page so he can answer them.

## Acceptance Criteria
- Create the NPS component.
- The user can swipe horizontally to change the score.
[Figma](https://www.figma.com/file/i1AI0IzCWuJApX5CK0DmiW/Challenge---Web-App?type=design&node-id=70-410&t=apcSQ1nCik2OLC0u-0)
image

### PR DESCRIPTION
**Pull request by manh-t**
Close #24 

## What happened 👀

Create a new NPS component to prepare for the survey

## Insight 📝

We will set the average score as default.

## Proof Of Work 📹

https://github.com/manh-t/react-survey/assets/60863885/e5351e4e-8bf1-43d0-b89f-c9f4076daf01

### CODE REVIEW

**Comment by manh-t** [comment]
🚀 Deployed on https://64faecdce1e6fb5e60839923--manh-react-survey.netlify.app

**Comment by manh-t** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Messages</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:book:</td>
      <td>Congrats, coverage is good</td>
    </tr>
  </tbody>
</table>

## Coverage in All Files
File | Line Coverage | Statement Coverage | Function Coverage | Branch Coverage
---- | ------------: | -----------------: | ----------------: | --------------:
[src/types/question.ts](../blob/efc9651/src/types/question.ts) | (1/2) 50% | (1/2) 50% | (0/1) 0% | (0/2) 0%
[src/../TextArea/index.tsx](../blob/efc9651/src/components/TextArea/index.tsx) | (5/8) 63% | (5/8) 63% | (1/2) 50% | (0/2) 0%
[src/../AppSlider/index.tsx](../blob/efc9651/src/components/AppSlider/index.tsx) | (4/6) 67% | (4/6) 67% | (1/3) 33% | (3/3) 100%
[src/tests/error.ts](../blob/efc9651/src/tests/error.ts) | (2/3) 67% | (2/3) 67% | (1/2) 50% | (1/3) 33%
[src/../Rating/index.tsx](../blob/efc9651/src/components/Rating/index.tsx) | (18/22) 82% | (18/22) 82% | (7/7) 100% | (4/13) 31%
[src/../Content/index.tsx](../blob/efc9651/src/components/Dashboard/Content/index.tsx) | (14/15) 93% | (14/15) 93% | (7/7) 100% | (6/9) 67%
[src/helpers/deserializer.ts](../blob/efc9651/src/helpers/deserializer.ts) | (17/18) 94% | (17/18) 94% | (7/7) 100% | (10/12) 83%
[src/../MultiChoice/index.tsx](../blob/efc9651/src/components/MultiChoice/index.tsx) | (19/20) 95% | (19/20) 95% | (7/8) 88% | (7/8) 88%
[src/../v1/requestManager.ts](../blob/efc9651/src/lib/request/v1/requestManager.ts) | (19/20) 95% | (19/20) 95% | (6/6) 100% | (3/5) 60%
[src/App.tsx](../blob/efc9651/src/App.tsx) | (3/3) 100% | (3/3) 100% | (1/1) 100% | (0/0) NaN%
[src/index.tsx](../blob/efc9651/src/index.tsx) | (3/3) 100% | (3/3) 100% | (0/0) NaN% | (0/0) NaN%
[src/../Authentication/index.ts](../blob/efc9651/src/adapters/Authentication/index.ts) | (4/4) 100% | (4/4) 100% | (2/2) 100% | (0/0) NaN%
[src/adapters/Base/index.ts](../blob/efc9651/src/adapters/Base/index.ts) | (8/8) 100% | (8/8) 100% | (2/2) 100% | (2/2) 100%
[src/adapters/BaseAuth/index.ts](../blob/efc9651/src/adapters/BaseAuth/index.ts) | (10/10) 100% | (10/10) 100% | (4/4) 100% | (2/2) 100%
[src/adapters/Survey/index.ts](../blob/efc9651/src/adapters/Survey/index.ts) | (2/2) 100% | (2/2) 100% | (2/2) 100% | (0/0) NaN%
Other (43 more) | (289/289) 100% | (289/289) 100% | (103/105) 98% | (85/90) 94%
Total | (418/433) 97% | (418/433) 97% | (151/159) 95% | (123/151) 81%

<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against ff160bc9f0bc6c95628980708e94aadc86da9d4e
</p>

File: src/components/Nps/index.tsx [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+  const onScoreChanged = (index: number) => {
+    const selectedValues = items.slice(0, index + 1);
+    setCurrentScore(index);
+    onValuesChanged(selectedValues);
+  };
+
+  return (
+    <div className="flex justify-center" data-test-id={npsDataTestIds.base}>
+      <div className="flex flex-col w-[335px]">
+        <div className="flex flex-row justify-center h-[56px]">
```
Comment (nvminhtue) [inline src/components/Nps/index.tsx]:
```suggestion
        <div className="flex justify-center h-[56px]">
```

**Review by nvminhtue — APPROVED** [review]

File: src/components/Nps/index.test.tsx [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+      render(<Nps items={answers} onValuesChanged={(answers) => handleValuesChanged(answers)} />);
+
+      const nps = screen.getByTestId(npsDataTestIds.base);
+
+      act(() => {
+        within(nps).getByText(answers[5].text).click();
+      });
+
+      expect(within(nps).getByText(answers[0].text)).not.toHaveClass('opacity-50');
```
Comment (hoangmirs) [inline src/components/Nps/index.test.tsx:43]: Same comment as previous PR, what is the `opacity-50`?

**Review by hoangmirs — APPROVED** [review]

File: src/components/Nps/index.test.tsx (hunk 2 above)
Comment (manh-t) [reply src/components/Nps/index.test.tsx:43]: @hoangmirs added ✅

**Pull request merged**